### DISCUSSION TITLE
Payment Methods: Adding switch to enable admin handling only

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#6088] [Ideas]
If a site's currently using payment-module `A`, but decides to switch to payment-module `B` ... payment-module `A` is disabled so that new orders can't be placed.  Unfortunately, that disablement results in payment-module `A` being disabled both storefront and admin.

This could pose problems for a store if they've still got orders that need to be either captured, refunded or voided that were placed using payment-module `A`.

Perhaps payment-modules should provide a tri-state "Enable this" setting to disable both storefront/admin, enable both storefront/admin or disable storefront-only?

**Comment by drbyte** [comment]
In the short-term, one way to get around that is to create a Zone for the payment module, and restrict that module to only that zone. Then it won't be usable on the storefront unless the customer's checkout address is part of that "zone".
Not a particularly pretty or intuitive solution, but it works.

As for a longer-term solution, a third option makes sense. I'd prefer that any "only enabled in Admin" switch be named something like "Retired: available only for initiating refunds" or something to that end. 

(Doesn't have to mention "captures" or "voids", but can still "do" those. Voids typically must be done very soon, and captures have a limited time-window as well, different based on provider and store's industry category.)

And, in the case of modules that have web-hook handlers/listeners like PayPal etc where an IPN or other incoming notification may tell the store about updates to orders, that listener should still "work" for tracking those notifications too (whether for order-status updates or other notifications). However, if the module is being replace with another module by the same provider (eg: PayPal), probably only the new module should be actively "listening". So the handler would need to know which module to direct any incoming notification to.

Naturally, if the storeowner doesn't retain the configuration of that old module, or deletes the account in that module's provider, then obviously none of these limited-use flags will continue to work.

**Comment by lat9** [comment]
> ...
> As for a longer-term solution, a third option makes sense. I'd prefer that any "only enabled in Admin" switch be named something like "**Retired: available only for initiating refunds**" or something to that end.
> 
> ...

That's quite the mouthful.  Any suggestions for a shorter value for that configuration choice?

**Reply by drbyte** [reply]
Is "legacy mode" too geek-speak? 🤓

**Reply by drbyte** [reply]
Or just "Retired" maybe?

**Reply by lat9** [reply]
IMO, yes (to the legacy mode).  How about a setting of "Admin Only" (short and sweet) with a more-wordy addition to the setting's description to flesh out the usage?  For example:
```
Use the "Admin Only" setting if you are planning to remove this payment module but still have administrative actions to perform against orders placed with this module?
```

**Reply by drbyte** [reply]
I can live with that. 
I just felt that something like "Retired" leaves little ambiguity, but "Admin Only" requires looking up what exactly that means.
But "Admin Only" is fine.

**Reply by scottcwilson** [reply]
"No new orders" perhaps?

**Reply by scottcwilson** [reply]
In gold, the universal color of geezers.  :)

**Comment by lat9** [comment]
I'll go with a combo, adding a setting of `Retired` and updating the description to include
```
Use the "Retired" setting if you are planning to remove this payment module but still have administrative actions to perform against orders placed with this module.
```

**Reply by scottcwilson** [reply]
Maybe "against *older* orders" or "against *existing* orders" for further emphasis. 

This is a solid plan.  Thanks for getting ahead of this issue.
